src/Maths/Silk.NET.Maths/Scalar.Constants.cs

```diff
@@ -0,0 +1,316 @@
+        private const float FloatPi = 3.14159274f;
+        private const float FloatTau = 6.283185307f;
+
+        [MethodImpl(MaxOpt)]
+        static Scalar()
+        {
+            // This won't inline as nicely on platforms that aren't .NET 5, however there's no other way to yield the
+            // constant folding benefits that come with the fields being static readonly.
+            //
+            // We have used local functions elsewhere to get around this elsewhere, however there's no sane way we can
+            // do that with local functions.
+            //
+            // This will inline fine on .NET 5, though. See also: https://github.com/dotnet/runtime/issues/38106
+            if (typeof(T) == typeof(Half))
+            {
+                Epsilon = (T) (object) Half.Epsilon;
+                MaxValue = (T) (object) Half.MaxValue;
+                MinValue = (T) (object) Half.MinValue;
+                NaN = (T) (object) Half.NaN;
+                NegativeInfinity = (T) (object) Half.NegativeInfinity;
+                PositiveInfinity = (T) (object) Half.PositiveInfinity;
+                One = (T) (object) (Half) 1;
+                Two = (T) (object) (Half) 2;
+                MinusOne = (T) (object) (Half) (-1f);
+                MinusTwo = (T) (object) (Half) (-2f);
+                E = (T) (object) (Half) FloatE;
+                Pi = (T) (object) (Half) FloatPi;
+                Tau = (T) (object) (Half) FloatTau;
+            }
+            else if (typeof(T) == typeof(float))
+            {
+                Epsilon = (T) (object) float.Epsilon;
+                MaxValue = (T) (object) float.MaxValue;
+                MinValue = (T) (object) float.MinValue;
+                NaN = (T) (object) float.NaN;
+                NegativeInfinity = (T) (object) float.NegativeInfinity;
+                PositiveInfinity = (T) (object) float.PositiveInfinity;
+                One = (T) (object) 1f;
+                Two = (T) (object) 2f;
+                MinusOne = (T) (object) -1f;
+                MinusTwo = (T) (object) -2f;
+                E = (T) (object) FloatE;
+                Pi = (T) (object) FloatPi;
+                Tau = (T) (object) FloatTau;
+            }
+            else if (typeof(T) == typeof(double))
+            {
+                Epsilon = (T) (object) double.Epsilon;
+                MaxValue = (T) (object) double.MaxValue;
+                MinValue = (T) (object) double.MinValue;
+                NaN = (T) (object) double.NaN;
+                NegativeInfinity = (T) (object) double.NegativeInfinity;
+                PositiveInfinity = (T) (object) double.PositiveInfinity;
+                One = (T) (object) 1d;
+                Two = (T) (object) 2d;
+                MinusOne = (T) (object) -1d;
+                MinusTwo = (T) (object) -2d;
+                E = (T) (object) Math.E;
+                Pi = (T) (object) Math.PI;
+#if !NET5_0
+                Tau = Scalar.Multiply(Pi, Two);
```

Comment:
Did MathF.Tau not make it into .NET 5?

Reply:
idk didn't touch this file. Will investigate.

Reply:
Ah I apologise